Netcracker/KubeMarine
# MANOPD-87827 Ensure thirdparties management approach

### Description
* It is necessary to restrict usages of not supported Kubernetes patch versions.
* Introduce rules to manage third-parties and add validation of them to the management tool.

### Solution
* Added exception if attempt to install/use/upgrade not supported Kubernetes patch versions.
* Added the following rules to the third-parties management tool:
    * It is allowed to remove Kubernetes version only with all previous versions having the same or lower minor version.
    * Downgrade of plugins is not allowed during upgrade of Kubernetes.
* Improve exception handling process.

### Test Cases

**TestCase 1**

Steps:

1. Remove intermediate Kubernetes version from `kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml`
2. Run `scripts/thirdparties/sync.py`

Results:

| Before | After |
| ------ | ------ |
| Success  | Exception should be raised |

**TestCase 2**

Steps:

1. Upgrade any plugin for intermediate Kubernetes version in `kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml`, but leave higher versions of Kubernetes having the same plugin versions as-is.
2. Run `scripts/thirdparties/sync.py`

Results:

| Before | After |
| ------ | ------ |
| Success  | Exception should be raised |

### Checklist

#### Unit tests
* Fixed test_upgrade.py to support new validation and removed obsolete validation.
* Fixed and added tests in test_sync.py.
